core/modules/wsched/SchedulerBase.h

```diff
@@ -105,6 +122,17 @@ class SchedulerBase : public wcontrol::Scheduler {
     int _priorityNext; ///< Priority to use starting with the next chunk.
 
     std::atomic<int> _inFlight{0}; //< Number of Tasks running.
+
+private:
+    /// The true purpose of _userQuerycount is to track how many different UserQuery's are on the queue.
+    /// Number of Tasks for each UserQuery in the queue.
+    std::map<uint64_t, int> _userQueryCounts;
```

Comment:
Is this the global query ID we all keep talking about? If so, we need to 
talk on how to pass it to the SSI layer so we can keep track of the query 
there as well.

Andy

On Fri, 29 Apr 2016, John Gates wrote:

>> @@ -105,6 +122,17 @@ class SchedulerBase : public wcontrol::Scheduler {
>>      int _priorityNext; ///< Priority to use starting with the next chunk.
>>
>>      std::atomic<int> _inFlight{0}; //< Number of Tasks running.
>> +
>> +private:
>> +    /// The true purpose of _userQuerycount is to track how many different UserQuery's are on the queue.
>> +    /// Number of Tasks for each UserQuery in the queue.
>> +    std::map<uint64_t, int> _userQueryCounts;
>
> I changed them on the worker side, but qmeta:QueryId is needed for swig and python, so we need a ticket for removing that and cleaning up.
>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/lsst/qserv/pull/220/files/856bf25bb7aed922d9dc8ea78cb71835ec3bbaf2#r61621219